### PR TITLE
Network retry changes 

### PR DESCRIPTION
**Pull request by cacheung**
Tested Analytics connection for the  latest NetworkService changes ((https://github.com/adobe/aepsdk-core-android/pull/612)

Analytics network connection == null is in place, it already retries when the connection returns null.
The invalid Analytics Configuration and  base url will be retried as previous implementation.

Added malformed url unit test in this PR.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: code/analytics/src/main/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsHitProcessor.kt [hunk 1]
```diff
@@ -248,6 +249,11 @@ internal class AnalyticsHitProcessor(
      */
     private fun getAnalyticsBaseUrl(state: AnalyticsState): String? {
         if (!state.isAnalyticsConfigured) {
+            Log.debug(
+                AnalyticsConstants.LOG_TAG,
+                CLASS_NAME,
+                "getAnalyticsBaseUrl - The Analytics configuration for RSID or host is not ready, as they appear to be empty or null."
```
Comment (kevinlind) [inline code/analytics/src/main/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsHitProcessor.kt:255]: suggest rewording "getAnalyticsBaseUrl - The Analytics configuration for RSID or host is not found. RSID and host must not be null or empty."

**Review by emdobrin — APPROVED** [review]

File: code/analytics/src/main/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsHitProcessor.kt (hunk 1 above)
Comment (cacheung) [reply code/analytics/src/main/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsHitProcessor.kt]: I want to keep it as debug since the other similar loggings in this file are in debug level.

**Review by kevinlind — APPROVED** [review]

File: code/analytics/src/main/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsHitProcessor.kt (hunk 1 above)
Comment (cacheung) [reply code/analytics/src/main/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsHitProcessor.kt]: After discussion with the team, I am changing the logging for recoverable connections to Debug level.

**Review by kevinlind — APPROVED** [review]
